libexec/daps-xmlwellformed

```diff
@@ -212,23 +131,112 @@ def parse_cli(args=None) -> argparse.Namespace:
     return args
 
 
-if __name__ == "__main__":
+def unique(lst: list):
+    """Makes a list unique (preserves order)
+
+    :param lst: the list that needs to be unique
+    :return: the (maybe) reduced list with unique elements
+    """
+    seen = {}
+
+    for item in lst:
+        if item in seen: continue
+        seen[item] = 1
+        yield item
+
+
+def get_all_files(xmlfile: str, xinclude: bool):
+    """Returns a list of all files when enabling XInclude processing
+
+    :param xmlfile: filename of XML file
+    :param xinclude: Enable or disable XInclude processing
+    :return:
+    """
+    ns = etree.FunctionNamespace(SUSE_NS)
+    ns['exists'] = exists
+    ns['abspath'] = abspath
+
+    xifiles = []
+    notfound = []
+    problems = []
+    error = None
+
+    try:
+        tree = etree.parse(xmlfile, parser=XMLPARSER)
+        transform = etree.XSLT(XSLT)
+        if xinclude:
+            result = transform(tree)
+            for entry in transform.error_log:
+                msg = entry.message
+                if msg.startswith("WARN:"):
+                    _, f = msg.split(":")
+                    notfound.append(f)
+                else:
+                    problems.append(str(entry))
+            xifiles = str(result).rstrip().split("\n")
+
+    except ValueError as err:
+        error = err
+        rc = ReturnCode.value_err
+    except etree.XSLTApplyError as err:
+        # We search for "Cannot resolve URI <FILENAME>"
+        uri = err.args[0].split("URI ")[-1]
+        problems.append(uri)
+    except etree.XIncludeError as error:
+        error = err
+        rc = ReturnCode.xinclude_err
+    except etree.XMLSyntaxError as err:
+        error = err
+        rc = ReturnCode.syntax_error
+
+    if error:
+        print(error, file=sys.stderr)
+        sys.exit(rc.value)
+
+    return xifiles, notfound, problems
+
+
+def main() -> int:
+    """Main function
+
+    :return: error code
+    """
+    failures, successes = (0, 0)
+
     args = parse_cli()
-    endresult=0
-    failures=0
-    successes=0
+
     for xmlfile in args.xmlfiles:
```

Comment:
I think you need to check somewhere whether the input files exist.

Reply:
Ok, done. Checked for `OSError` exception.